test/replication-luatest/gh_5295_split_brain_test.lua

```diff
@@ -0,0 +1,247 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+
+local g = t.group('gh-6036')
+
+local function repl_list(i)
+    local repl = {}
+    for j = 1,i do
+        table.insert(repl, server.build_instance_uri('s'..j))
+    end
+    return repl
+end
+
+-- gh-5295: the server should stop replication from an upstream which sends data
+-- conflicting in terms of promote / confirmed_lsn.
+--
+-- The test requires one instance in replicaset per each split-brain case, plus
+-- one "main" instance. The idea is to partition one instance from the main
+-- server per test case. Once partitioned, create a split-brain situation
+-- between the partitioned node and all the main's followers. Then check that
+-- the partitioned node can't reconnect.
+local num_instances = 1
+
+g.before_all(function(cg)
+    cg.cluster = cluster:new({})
+
+    cg.box_cfg = {
+        replication = repl_list(num_instances),
+        replication_timeout         = 0.1,
+        replication_synchro_quorum  = 1,
+        replication_synchro_timeout = 0.01,
+        election_mode               = 'manual',
+        election_timeout            = 0.1,
+        election_fencing_enabled    = false,
+        log_level                   = 6,
+    }
+
+    cg.s = {}
+    for i = 1,num_instances do
```

Comment:
Lets scatter some whitespaces after commas in this file.

Reply:
As you wish. I thought `for j = 1,i` was an idiom in lua.